### ISSUE TITLE
Can't start RLS – detected channel name includes default

### ISSUE THREAD

**Issue by danlucraft** [rust-lang/rls-vscode#522]
It appears that rls-vscode is detecting a channel name like "stable-x86_64-apple-darwin (default)":
```
[Extension Host] Detected active channel: stable-x86_64-apple-darwin (default) (since 'rust-client.channel' is unspecified)
```
This is causing the rustup command to fail as the "(default)" part is not actually part of the toolchain name:

```
Error: Command failed: rustup component list --toolchain stable-x86_64-apple-darwin (default)
/bin/sh: -c: line 0: syntax error near unexpected token `('
/bin/sh: -c: line 0: `rustup component list --toolchain stable-x86_64-apple-darwin (default)'
```

Stripping the "(default)" and setting `rust-client.channel` to "stable-x86_64-apple-darwin" fixes the issue.

So it seems like the code that detects the channel name needs to be able to recognise and strip "(default)" from wherever it is finding that. I'll take a look and see if I can see where that happens, but I'm not experienced with VSCode extensions, or rustup.

```
vscode  1.32.1
rustc   1.33.0 (2aa4c46cf 2019-02-28)
rustup  1.17.0 (069c88ed6 2019-03-05)
```

**Comment by danlucraft** [comment]
Aaaaaand closing this because I found the closed issues that discuss it.

**Comment by Xanewok** [comment]
Pinning it to make it more visible for other folks; this should be fixed with https://github.com/rust-lang/rls-vscode/pull/520.

**Comment by softprops** [comment]
> Pinning it to make it more visible for other folks

Is there a work around atm?

**Comment by angusgraham** [comment]
@softprops yes, the solution is in https://github.com/rust-lang/rls-vscode/issues/517:

Add "rust-client.channel": "stable" to your settings.

**Comment by angusgraham** [comment]
Fixed in 0.5.4
